File: python/70.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        table = [0] * n
        # base case step 1
        table[0] = 1
        for i in range (1, n):
            # base case step 2
            if i == 1:
                table[1] = 2
            else:
                # for the i-th step, we either take 1 step from step i - 1
                #   or 2 steps from step i - 2
                table[i] = table[i - 2] + table[i - 1]
        return table[-1]

File: c/70.c
int climbStairs(int n) {
    // base cases
    if (n == 1){
        return 1;
    }
    if (n == 2){
        return 2;
    }
    // 45 since by constraints, n <= 45
    int dp[45] = {};
    for (int i=0; i < 45; i++){
        dp[i] = 0;
    }
    dp[0] = 1;
    dp[1] = 2;
    // for the i-th step, we either take 1 step from step i - 1
    //   or 2 steps from step i - 2
    for (int i = 2; i < n; i++){
        dp[i] = dp[i - 1] + dp[i - 2];
    }
    return dp[n - 1];
}
